grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -374,7 +373,7 @@ class QueryService implements Status {
             setParameterList('consentCollectionIds', consentCollectionIds)
             list()
         }
-        results.groupBy { it.consentCollectionLinkId as BigInteger }
+        results.groupBy { it.consentCollectionLinkId }
```

Comment:
Yes, exactly ... this is much cleaner! Using `Object`, `def`, and casting are obvious [code-smells](https://martinfowler.com/bliki/CodeSmell.html) that indicate there is probably a better way of doing what you want.